netplugin/netd.go

```diff
@@ -14,248 +14,295 @@ limitations under the License.
+		logrus.Infof("Netplugin exiting due to error: %v", err)
+		os.Exit(1)
 	}
+}
 
-	if opts.version {
-		fmt.Printf(version.String())
-		os.Exit(0)
-	}
+func validateConfigs(ctx *cli.Context) (plugin.Config, error) {
 
-	// Make sure we are running as root
-	usr, err := user.Current()
-	if (err != nil) || (usr.Username != "root") {
-		log.Fatalf("This process can only be run as root")
+	// 1. validate and set up log
+	if ctx.Bool("use-syslog") {
+		syslogURL := ctx.String("syslog-url")
+		configureSyslog(syslogURL)
+		logrus.Infof("Set netplugin use syslog config: %v", syslogURL)
+	} else {
+		logrus.Infof("Set netplugin use syslog config: false")
 	}
 
-	if opts.debug {
-		log.SetLevel(log.DebugLevel)
-		os.Setenv("CONTIV_TRACE", "1")
+	logLevel, err := logrus.ParseLevel(ctx.String("log-level"))
+	if err != nil {
+		return plugin.Config{}, err
 	}
+	logrus.SetLevel(logLevel)
+	logrus.Infof("Use netplugin log level: %v", logLevel)
 
-	if opts.jsonLog {
-		log.SetFormatter(&log.JSONFormatter{})
+	if ctx.Bool("use-json-log") {
+		logrus.SetFormatter(&logrus.JSONFormatter{})
+		logrus.Infof("Use netplugin log format: json")
 	} else {
-		log.SetFormatter(&log.TextFormatter{FullTimestamp: true, TimestampFormat: time.StampNano})
+		logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true, TimestampFormat: time.StampNano})
+		logrus.Infof("Use netplugin log format: text")
 	}
 
-	if opts.syslog != "" {
-		configureSyslog(opts.syslog)
+	// 2. validate and set plugin mode
+	pluginMode := strings.ToLower(ctx.String("mode"))
+	switch pluginMode {
+	case master.Docker, master.Kubernetes, master.SwarmMode, master.Test:
+		logrus.Infof("Use netplugin mode: %v", pluginMode)
+	case "":
+		return plugin.Config{}, fmt.Errorf("netplugin mode is not set")
+	default:
+		return plugin.Config{}, fmt.Errorf("Unknown netplugin mode: %v", pluginMode)
 	}
 
-	if flagSet.NFlag() < 1 {
-		log.Infof("host-label not specified, using default (%s)", opts.hostLabel)
+	// 3. validate and set network mode
+	networkMode := strings.ToLower(ctx.String("netmode"))
+	switch networkMode {
+	case "vlan", "vxlan":
+		logrus.Infof("Use netplugin network mode: %v", networkMode)
+	case "":
+		return plugin.Config{}, fmt.Errorf("netplugin network mode is not set")
+	default:
+		return plugin.Config{}, fmt.Errorf("Unknown netplugin network mode: %v", networkMode)
 	}
 
-	// default to using local IP addr
-	localIP, err := cluster.GetLocalAddr()
-	if err != nil {
-		log.Fatalf("Error getting local address. Err: %v", err)
+	// 4. validate forward mode
+	forwardMode := strings.ToLower(ctx.String("fwdmode"))
+	if (networkMode == "vlan" && forwardMode == "bridge") || forwardMode == "routing" {
+		logrus.Infof("Use netplugin forwarding mode: %v", forwardMode)
+	} else if forwardMode == "" {
+		return plugin.Config{}, fmt.Errorf("netplugin network mode is not set")
+	} else {
+		return plugin.Config{}, fmt.Errorf("Invalid netplugin forwarding mode: %v", forwardMode)
```

Comment:
maybe needs informative context when bridge but vxlan

Reply:
the netmode was print out as log the one above, supported combination should be in document

Reply:
that's frustrating to end user to decipher why it would say
`invalid netplugin forwarding mode: bridge`
when of course bridge is valid, but not with vxlan .. vs pinpointing `bridge forwarding mode is invalid with vxlan` and `invalid netplugin forwarding mode: something_really_wrong`